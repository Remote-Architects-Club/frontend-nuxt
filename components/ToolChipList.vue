<template>
  <div>
    <ul v-if="tools.length > 0" class="flex flex-wrap">
      <li v-for="tool in tools" :key="tool.id" class="mb-2 mr-1">
        <tool-chip :id="tool.id" :name="tool.name" :url="tool.url" />
      </li>
    </ul>
  </div>
</template>

<script>
import ToolChip from '@/components/ToolChip'
export default {
  name: 'ToolChipList',
  components: {
    ToolChip
  },
  props: {
    tools: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style lang="scss" scoped></style>
