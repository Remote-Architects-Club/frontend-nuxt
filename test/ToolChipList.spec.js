import { mount, RouterLinkStub } from '@vue/test-utils'
import ToolChipList from '@/components/ToolChipList.vue'

const factory = ({ propsData }) =>
  mount(ToolChipList, {
    propsData,
    stubs: {
      NuxtLink: RouterLinkStub
    }
  })

describe('ToolChipList', () => {
  test('is a Vue instance', () => {
    const wrapper = factory({
      propsData: {
        tools: [
          { id: 'id1', name: 'name1', url: 'http://url1.com' },
          { id: 'id2', name: 'name2', url: 'http://url2.com' },
          { id: 'id3', name: 'name3', url: 'http://url3.com' }
        ]
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('renders properly when 3 items are present', () => {
    const wrapper = factory({
      propsData: {
        tools: [
          { id: 'id1', name: 'name1', url: 'http://url1.com' },
          { id: 'id2', name: 'name2', url: 'http://url2.com' },
          { id: 'id3', name: 'name3', url: 'http://url3.com' }
        ]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('renders properly when 1 items is present', () => {
    const wrapper = factory({
      propsData: {
        tools: [{ id: 'id1', name: 'name1', url: 'http://url1.com' }]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('renders properly when an empty list is passed as prop', () => {
    const wrapper = factory({
      propsData: {
        tools: []
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('renders properly when prop is not present', () => {
    const wrapper = factory({ propsData: {} })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
