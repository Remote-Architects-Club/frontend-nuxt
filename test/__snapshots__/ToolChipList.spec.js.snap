// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolChipList renders properly when 1 items is present 1`] = `
"<div>
  <ul class=\\"flex flex-wrap\\">
    <li class=\\"mb-2 mr-1\\">
      <div><a data-cy=\\"tool-name\\" class=\\"px-4 py-1 border border-black rounded-full hover:bg-yellow-500 hover:cursor-pointer\\">name1</a></div>
    </li>
  </ul>
</div>"
`;

exports[`ToolChipList renders properly when 3 items are present 1`] = `
"<div>
  <ul class=\\"flex flex-wrap\\">
    <li class=\\"mb-2 mr-1\\">
      <div><a data-cy=\\"tool-name\\" class=\\"px-4 py-1 border border-black rounded-full hover:bg-yellow-500 hover:cursor-pointer\\">name1</a></div>
    </li>
    <li class=\\"mb-2 mr-1\\">
      <div><a data-cy=\\"tool-name\\" class=\\"px-4 py-1 border border-black rounded-full hover:bg-yellow-500 hover:cursor-pointer\\">name2</a></div>
    </li>
    <li class=\\"mb-2 mr-1\\">
      <div><a data-cy=\\"tool-name\\" class=\\"px-4 py-1 border border-black rounded-full hover:bg-yellow-500 hover:cursor-pointer\\">name3</a></div>
    </li>
  </ul>
</div>"
`;

exports[`ToolChipList renders properly when an empty list is passed as prop 1`] = `
"<div>
  <!---->
</div>"
`;

exports[`ToolChipList renders properly when prop is not present 1`] = `
"<div>
  <!---->
</div>"
`;
